File: c/prob2.c
#include <stdio.h>

int main()
{
	int num = 1;
	int last = 1;
	int sum = 0;
	while(num <= 4000000)
	{
		if(num%2==0)sum += num;
		num += last;
		last = num - last;
	}

	printf("%d\n",sum);
}

File: perl/prob2.pl
#!/usr/bin/env perl

$last = 0;
$curr = 1;
$sum = 0;
while($curr <= 4000000)
{
	$curr += $last;
	$last = $curr - $last;
	$sum += $curr if $curr % 2 ==0;
}
print $sum . "\n";

